File: androidApp/src/main/java/com/example/coffeeshop/android/screen/splash_screen/SplashViewModel.kt
package com.example.coffeeshop.android.screen.splash_screen

import android.util.Log
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.coffeeshop.android.untils.Constants.AUTH_NOT_STATED
import com.example.coffeeshop.android.untils.Constants.NOT_AUTH
import com.example.coffeeshop.android.untils.Constants.AUTH_SUCCESS
import com.example.coffeeshop.di.UseCases
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class SplashViewModel @Inject constructor(
):ViewModel(){

    private val _authUIState = MutableStateFlow(AUTH_NOT_STATED)
    val authUIState:StateFlow<Int> = _authUIState

    fun init(){
        viewModelScope.launch {
            val user = UseCases.useGetUserData().execute().data

            Log.d("sdfsdfffddff", user.toString())

            if(user != null){
                UseCases.useSyncCoffeeCategories().execute()
                UseCases.useSyncCoffee().execute()
                UseCases.useSyncUserData().execute()
                UseCases.useSyncOrders().execute()
            }

            _authUIState.value =  if(user == null) NOT_AUTH else AUTH_SUCCESS
        }
    }

}

File: iosApp/iosApp/splash_screen/SplashViewModel.swift
//
//  SplashViewModel.swift
//  iosApp
//
//  Created by MacBook on 13.12.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import shared

class SplashViewModel:ObservableObject{
    
    @Published var auth_succeed = false
    @Published var auth_failure = false
    
    init() {
        UseCases().useGetUserData().execute(completionHandler: {user, err in
            if(user?.data != nil){
                UseCases().useSyncCoffeeCategories().execute(completionHandler: {_,_ in
                    UseCases().useSyncCoffee().execute(completionHandler: {_,_ in
                        UseCases().useSyncUserData().execute(completionHandler: {_,_ in
                            UseCases().useSyncOrders().execute(completionHandler: {_,_ in
                                self.auth_succeed = true
                            })
                        })

                    })

                })
            }else{
                self.auth_failure = true
            }
        })
    }
    
}
